src/resources/helpers/generate-method.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const { WorkastInvalidParameterError } = require('../../errors');
+
+/**
+ * Method spec.
+ *
+ * @typedef {Object} MethodSpec
+ * @property {string} method - The HTTP method. One of 'GET', 'HEAD', 'POST', 'PUT', 'DELETE' or 'PATCH'.
+ * @property {string} path - The generic URL path.
+ * */
+
+/**
+ * @summary Generates an API method from the given spec.
+ *
+ * @param {Workast} workast - The Workast SDK instance.
+ * @param {MethodSpec} spec - The method spec.
+ *
+ * @returns {Function} An async function to call the API endpoint.
+ * */
+function generateMethod(workast, spec) {
+  // eslint-disable-next-line func-names
+  return async function (...args) {
+    const urlParamsCount = (spec.path.match(/{\w+}/g) || []).length;
+
+    if (args.length < urlParamsCount) {
+      throw new WorkastInvalidParameterError('Missing URL params');
+    }
+
+    // Build the actual path.
+    let { path } = spec;
+
+    for (let i = 0; i < urlParamsCount; i += 1) {
+      path = path.replace(/{\w+}/, args[i]);
+    }
+
+    // Determine if body or query should be used.
+    let query;
+    let body;
+
+    if (['GET', 'HEAD'].includes(spec.method)) {
```

Comment:
We are generating this array each time the function is executed in runtime, we should have this in the parent function and make use of the closure setup with something like `const queryParamsMethods = ['GET', 'HEAD']` in line 22 and access it inside the function.

Reply:
Yes, that could work.